azkaban-web-server/src/restli/java/azkaban/restli/ProjectManagerResource.java

```diff
@@ -111,16 +120,24 @@ public String deploy(@ActionParam("sessionId") final String sessionId,
     }
 
     try {
+      if (enableQuartz) {
+        scheduler.unscheduleAll(project);
+      }
       // Check if project upload runs into any errors, such as the file
       // having blacklisted jars
       final Props props = new Props();
       final Map<String, ValidationReport> reports = projectManager
           .uploadProject(project, archiveFile, "zip", user, props);
+
```

Comment:
If it throws exception at uploadProject step, the existing schedules will still be removed but not added back?

Reply:
it's a good point. Actually it could happen when HDFS as project zip storage layer is down but mysql is up. Schedule is not the only thing that worth concerning if this is the case. Maintaining atomicity is complicated enough to fit into another PR. added todo here for record.